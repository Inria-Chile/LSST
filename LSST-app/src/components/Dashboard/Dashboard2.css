#dashboard {
    height: 100vh;
    width: 100vw;
    overflow: hidden;
}

#dashboard-content {
    display: inline-block;
}

#dashboard-toolbar{
    height: 100%;
    background: #2d4053;
    display: inline-block;    
}

#dashboard-content > div {
    background: #2d4053;
    margin: 2px;
    cursor: default;
}

#toolbar {
    width: 3vw;
}

.field-details-popup {
    box-shadow: 5px 5px 20px rgba(0,0,0,0.5);    
}

.react-draggable {
    overflow: hidden;
    box-shadow: 0px 0px 15px rgba(0,0,0,0.3);
}

.react-draggable-dragging {
    overflow: hidden;
    box-shadow: 5px 5px 30px black;
    transition: box-shadow 0.3s ease-in-out;
    z-index: 100;
}

.field-details-popup.react-draggable-dragging {
    overflow: hidden;
    box-shadow: 10px 10px 60px black;
    transition: box-shadow 0.3s ease-in-out;
    z-index: 100;
}