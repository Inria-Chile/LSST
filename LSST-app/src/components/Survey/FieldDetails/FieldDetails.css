.histogram .bar[fill='']{
    fill: var(--theme-histogram-bar)
}

.histogram .bar[fill='latest']{
    fill: var(--theme-histogram-bar-latest);
}

.histogram {
    padding: 20px 50px;
}

.histogram .tick {   
    stroke: var(--theme-inactive-text);
}

.histogram text {   
    fill: var(--theme-inactive-text);
    font-size: 13px;
}

.histogram path {   
    stroke: var(--theme-inactive-text);
    stroke-opacity: 1;
}

.histogram line {   
    stroke: var(--theme-inactive-text);
}

.barchart {
    padding: 0px 0px;
}

.field-details {
    background: var(--theme-div-background);
    padding-bottom: 20px;
    height: 100%;
}

.field-details-content {
    display: flex;
    flex-wrap: wrap;
}

.field-details-title, .field-details-title h2 {
    background: var(--theme-background);
    margin: 0px;
    padding: 5px;
}

.barchart .axis text {
    font-size: 13px;
    font-weight: normal;
}

.field-details-content h5 {
    text-align: center;
    color: var(--theme-inactive-text);
}

.field-details-content > div {
    margin: 10px;
    width: 370px;
}

.field-details-content > div > div {
    background: var(--theme-inactive-slider);
}

.summary-data-content {
    padding: 30px;
}