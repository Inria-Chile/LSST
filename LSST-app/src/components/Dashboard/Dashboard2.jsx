import React, { Component } from 'react';
// import ReactDOM from 'react-dom';
import Survey from '../Survey/Survey';
import Mirrors from '../Mirrors/Mirrors';
import Dome from '../Dome/Dome';
import Environment from '../Environment/Environment';
import FieldDetails from '../Survey/FieldDetails/FieldDetails';
import {Responsive, WidthProvider} from 'react-grid-layout';
import './Dashboard2.css';
const ReactGridLayout = WidthProvider(Responsive);

class Dashboard2 extends Component {
    
    static viewComponents = [Survey, Mirrors];
    static layout = {
        'survey': {i: 'survey', x: 0, y: 0, w: 6, h: 10},
        'mirrors': {i: 'mirrors', x: 6, y: 0, w: 2, h: 12},
        'dome': {i: 'dome', x: 8, y: 0, w: 4, h: 20},
        'environment': {i: 'environment', x: 0, y: 10, w: 3, h: 3}
    };
    constructor(props){
        super(props);
        this.state = {
            showFieldDetails: false,
            environmentOpenRows: 0,
            layout: Object.values(Dashboard2.layout),
        }
    }

    setFieldDetailsVisibility = (visibility) => {
        this.setState({
            showFieldDetails: visibility,
        });
    }

    setSelectedFieldData = (data) => {
        this.setState({
            selectedFieldData: data
        });
    }

    setComponentHeight = (component, height) => {
        let newLayout = JSON.parse(JSON.stringify(this.state.layout));
        newLayout.map((l, i) => {
            if(l.i === component)
                newLayout[i] = {...newLayout[i], h:height};
            return l;
        });
        this.setState({
            layout: newLayout
        });
    }

    setEnvironmentOpenRows = (rows) => {
        this.setComponentHeight('environment', 3+rows);
    }

    onLayoutChange = (layout) => {
        this.setState({layout: layout});
    }

    render() {
        return (
            <div id="dashboard" ref={(dashboard) => {
                if(!this.state.dashboardNode){
                    this.setState({
                        dashboardNode: dashboard
                    })
                }
            }}>
                <div id="dashboard-content2">
                    <ReactGridLayout className="layout" layouts={{lg: this.state.layout}} rowHeight={90} width={3840}
                        breakpoints={{lg: 3200, md: 996, sm: 768, xs: 480, xxs: 0}} draggableHandle={'.move-button'}
                        cols={{lg: 12, md: 6, sm: 6, xs: 6, xxs: 6}} onLayoutChange={this.onLayoutChange}>

                        <div key="survey">
                            <Survey parentNode={this.state.dashboardNode} 
                                    setFieldDetailsVisibility={this.setFieldDetailsVisibility} 
                                    showFieldDetails={this.state.showFieldDetails}
                                    setSelectedFieldData={this.setSelectedFieldData}/>
                        </div>
                        <div key="mirrors">
                            <Mirrors />
                        </div>
                        <div key="dome">
                            <Dome />
                        </div>
                        <div key="environment">
                            <Environment setOpenRows={this.setEnvironmentOpenRows}/>
                        </div>
                    </ReactGridLayout>
                </div>
                    {
                        this.state.showFieldDetails ? 
                        <FieldDetails fieldData={this.state.selectedFieldData}
                                        setFieldDetailsVisibility={this.setFieldDetailsVisibility}
                                        showFieldDetails={this.state.showFieldDetails}/>
                        :
                        ''
                    }
            </div>
        );
    }
}

export default Dashboard2;
