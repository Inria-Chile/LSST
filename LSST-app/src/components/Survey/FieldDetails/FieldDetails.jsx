import React, { PureComponent } from 'react'
import './FieldDetails.css';
import { BarChart } from 'react-d3-components'
import GenericHistogram from './GenericHistogram/GenericHistogram'
import Rnd from 'react-rnd';
import DraggableTitle from '../../Utils/DraggableTitle';
import FilterIndicator from '../../Utils/FilterIndicator/FilterIndicator';
import {lsstToJs} from '../../Utils/Utils'

class FieldDetails extends PureComponent {
    
    // - total number of visits
    // - date/time of last visit
    // DONE - histograms with number of visits in each filter
    // DONE - histogram of rotation angle of observations
    // DONE - airmass histogram 
    //   (highlight most recent/current observation with a vertical line)
    // DONE - histogram of seeing values 
    //   (highlight most recent/current observation with a vertical line)
    // DONE - sky brightness histogram 
    // DONE - histogram of time baselines (i.e., time between visits)
    // - histogram of number of alerts issued for this field (moving, variable, outburst, new , vanished)

    // Which sky brightness: VskyBright, perry_skybrightness, skybrightness_modified
    // Which rotation angle: rotSkyPos, rotTelPos
    // Which seeing: seeing, maxSeeing, cldSeeing
    // Baseline histogram: Confirm meaning, find solution to

    constructor(props){
        super(props);
        this.alarmsData = [{
            label: 'somethingA',
            values: [{x: 'Moving', y: 10}, {x: 'Variable', y: 4}, {x: 'Outburst', y: 1}, {x: 'New', y: 5}, {x: 'Vanished', y: 9}]
        }];
    }

    getBaseLines(data) {
        let baselines = [];
        for(let i=1;i<data.length;++i){
            if(data[i-1].expDate === data[i].expDate)
                continue;
            baselines.push(data[i-1].expDate - data[i].expDate)
        }
        return baselines;
    }

    render() {
        let width = 350;
        let height = 190;
        let fieldData = this.props.fieldData && this.props.fieldData.length > 0 ? this.props.fieldData : [{fieldID: 0}];
        let filtersCount = {'u':0, 'g':0, 'r':0, 'i':0, 'z':0, 'y':0};
        fieldData.map( x => {
            if(x['filterName'] !== undefined)
                filtersCount[x['filterName']]++;
            return 0;
        });
        let rotationData = fieldData.map( x => Math.round(x['rotSkyPos']*180/Math.PI));
        let airmassData = fieldData.map( x => x['airmass']);
        let seeingData = fieldData.map( x => x['seeing']);
        let skyBrightnessData = fieldData.map( x => x['skybrightness_modified']);
        let baselinesData = this.getBaseLines(fieldData);

        let filtersData = [{
            label: 'dsaad',
            values: Object.keys(filtersCount).map( f => {return {x: f, y: filtersCount[f]}})
        }];
        let date = new Date(lsstToJs(fieldData[0].expDate));

        return (
            <Rnd default={{
                x: 800,
                y: 100,
                width: 1560,
                height: 580,
                }}
                dragGrid={ [20,20]}
                resizeGrid={ [20,20]}
                disableDragging={false}
                enableResizing={{ top:true, right:true, bottom:true, left:true, topRight:true, bottomRight:true, bottomLeft:true, topLeft:true }}
                className={'field-details-popup'}
                dragHandleClassName={'.move-button'}
                resizeHandleClassName={'.close-button'}
                resizeHandleWrapperClass={'field-details-popup'}
            >
                <div className="field-details">
                    <DraggableTitle title={'Field ID: ' + fieldData[0].fieldID} customClass='field-details-title' showClose={true} onCloseClick={() => this.props.setFieldDetailsVisibility(false)}/>
                    <div className='field-details-content'>
                        <div className='summary-data'>
                            <h5>Summary</h5>
                            <div className='summary-data-content'>
                                <div>FieldID: {fieldData[0].fieldID}</div>
                                <div>Timestamp: {date.toDateString() + ' ' + date.toLocaleTimeString()}</div>
                                <div>Filter: <FilterIndicator filterName={fieldData[0].filterName}/></div>
                                <div>Observations: {fieldData.length}</div>
                            </div>
                        </div>
                        <div className=''>
                            <h5>Rotation angle</h5>
                            <GenericHistogram id='rotation-hist' data={rotationData} width={width-40} height={height-40} margin={60} xLabel={'Rotation [deg]'} yLabel={'Count'} domain={[0, 360]} nBins={36}  nTicks={5}/>
                        </div>
                        <div className='histogram barchart'>
                            <h5>Filters</h5>
                            <BarChart data={filtersData} width={width} height={height} margin={{top: 20, bottom: 30, left: 30, right: 20}} yAxis={{label: 'Count', tickArguments:[5]}}/>
                        </div>
                        <div className='histogram barchart'>
                            <h5>Alerts</h5>
                            <BarChart data={this.alarmsData} width={width} height={height} margin={{top: 20, bottom: 30, left: 30, right: 20}} yAxis={{label: 'Count', tickArguments:[5]}}/>
                        </div>
                        <div className=''>
                            <h5>Airmass</h5>
                            <GenericHistogram id='airmass-hist' data={airmassData} width={width-40} height={height-40} margin={60} xLabel={'Airmass [?]'} yLabel={'Count'} domain={[0, 3]} nBins={36}  nTicks={5}/>
                        </div>
                        <div className=''>
                            <h5>Seeing</h5>
                            <GenericHistogram id='seeing-hist' data={seeingData} width={width-40} height={height-40} margin={60} xLabel={'Seeing [?]'} yLabel={'Count'} domain={[0, 2]} nBins={36} nTicks={5}/>
                        </div>
                        <div className=''>
                            <h5>Sky brightness</h5>
                            <GenericHistogram id='sky-brightness-hist' data={skyBrightnessData} width={width-40} height={height-40} margin={60} xLabel={'Sky brightness [?]'} yLabel={'Count'} nBins={36} nTicks={5}/>
                        </div>
                        <div className=''>
                            <h5>Time baselines</h5>
                            <GenericHistogram id='time-baselines-hist' data={baselinesData} width={width-40} height={height-40} margin={60} xLabel={'Time baseline [s]'} yLabel={'Count'} nBins={36} nTicks={5} logScale={true}/>
                        </div>
                    </div>
                </div>
            </Rnd>
            );
       
    }
}

export default FieldDetails;