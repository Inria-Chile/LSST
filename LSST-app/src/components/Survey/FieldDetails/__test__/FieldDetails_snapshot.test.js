import React from 'react';
import FieldDetails from '../FieldDetails';
import renderer from 'react-test-renderer';
import shallowRenderer from 'react-test-renderer/shallow';
import Dashboard2 from '../../../Dashboard/Dashboard2';

describe('Fields details test',function(){
    it('renders correctly with shallowRender',()=>{
        const renderer = new shallowRenderer();
        let fieldData = {
            fieldID: 15, 
            filterName : 'y',
            expDate : "2018-01-04T19:44:10.611Z",
        };
        const tree = renderer.render( 
        <FieldDetails fieldData={[fieldData]}
            setFieldDetailsVisibility={Dashboard2.setFieldDetailsVisibility}
            showFieldDetails={true}/>)
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with shallowRender without fieldData',()=>{
        const renderer = new shallowRenderer();
        let fieldData = {
            fieldId: 15, 
            filterName : 'y',
            expDate : "2018-01-04T19:44:10.611Z",
        };
        const tree = renderer.render( 
        <FieldDetails 
            setFieldDetailsVisibility={Dashboard2.setFieldDetailsVisibility}
            showFieldDetails={true}/>)
        expect(tree).toMatchSnapshot();
    });


});