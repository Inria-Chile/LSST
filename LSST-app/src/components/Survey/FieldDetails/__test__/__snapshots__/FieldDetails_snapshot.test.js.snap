// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields details test renders correctly with shallowRender 1`] = `
<Rnd
  className="field-details-popup"
  default={
    Object {
      "height": 580,
      "width": 1560,
      "x": 800,
      "y": 100,
    }
  }
  disableDragging={false}
  dragGrid={
    Array [
      20,
      20,
    ]
  }
  dragHandleClassName=".move-button"
  enableResizing={
    Object {
      "bottom": true,
      "bottomLeft": true,
      "bottomRight": true,
      "left": true,
      "right": true,
      "top": true,
      "topLeft": true,
      "topRight": true,
    }
  }
  maxHeight={9007199254740991}
  maxWidth={9007199254740991}
  onDrag={[Function]}
  onDragStart={[Function]}
  onDragStop={[Function]}
  onResize={[Function]}
  onResizeStart={[Function]}
  onResizeStop={[Function]}
  resizeGrid={
    Array [
      20,
      20,
    ]
  }
  resizeHandleClassName=".close-button"
  resizeHandleWrapperClass="field-details-popup"
>
  <div
    className="field-details"
  >
    <DraggableTitle
      customClass="field-details-title"
      onCloseClick={[Function]}
      showClose={true}
      title="Field ID: 15"
    />
    <div
      className="field-details-content"
    >
      <div
        className="summary-data"
      >
        <h5>
          Summary
        </h5>
        <div
          className="summary-data-content"
        >
          <div>
            FieldID: 
            15
          </div>
          <div>
            Timestamp: 
            Invalid Date Invalid Date
          </div>
          <div>
            Filter: 
            <FilterIndicator
              filterName="y"
            />
          </div>
          <div>
            Observations: 
            1
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <h5>
          Rotation angle
        </h5>
        <GenericHistogram
          data={
            Array [
              NaN,
            ]
          }
          domain={
            Array [
              0,
              360,
            ]
          }
          height={150}
          id="rotation-hist"
          margin={60}
          nBins={36}
          nTicks={5}
          width={310}
          xLabel="Rotation [deg]"
          yLabel="Count"
        />
      </div>
      <div
        className="histogram barchart"
      >
        <h5>
          Filters
        </h5>
        <Unknown
          barPadding={0.5}
          colorByLabel={false}
          colorScale={[Function]}
          data={
            Array [
              Object {
                "label": "filters",
                "values": Array [
                  Object {
                    "x": "u",
                    "y": 0,
                  },
                  Object {
                    "x": "g",
                    "y": 0,
                  },
                  Object {
                    "x": "r",
                    "y": 0,
                  },
                  Object {
                    "x": "i",
                    "y": 0,
                  },
                  Object {
                    "x": "z",
                    "y": 0,
                  },
                  Object {
                    "x": "y",
                    "y": 1,
                  },
                ],
              },
            ]
          }
          height={190}
          label={[Function]}
          margin={
            Object {
              "bottom": 30,
              "left": 30,
              "right": 20,
              "top": 20,
            }
          }
          offset="zero"
          order="default"
          tooltipContained={false}
          tooltipHtml={null}
          tooltipMode="mouse"
          tooltipOffset={
            Object {
              "left": 0,
              "top": -35,
            }
          }
          values={[Function]}
          width={350}
          x={[Function]}
          xScale={null}
          y={[Function]}
          y0={[Function]}
          yAxis={
            Object {
              "label": "Count",
              "tickArguments": Array [
                5,
              ],
            }
          }
          yScale={null}
        />
      </div>
      <div
        className="histogram barchart"
      >
        <h5>
          Alerts
        </h5>
        <Unknown
          barPadding={0.5}
          colorByLabel={false}
          colorScale={[Function]}
          data={
            Array [
              Object {
                "label": "somethingA",
                "values": Array [
                  Object {
                    "x": "Moving",
                    "y": 10,
                  },
                  Object {
                    "x": "Variable",
                    "y": 4,
                  },
                  Object {
                    "x": "Outburst",
                    "y": 1,
                  },
                  Object {
                    "x": "New",
                    "y": 5,
                  },
                  Object {
                    "x": "Vanished",
                    "y": 9,
                  },
                ],
              },
            ]
          }
          height={190}
          label={[Function]}
          margin={
            Object {
              "bottom": 30,
              "left": 30,
              "right": 20,
              "top": 20,
            }
          }
          offset="zero"
          order="default"
          tooltipContained={false}
          tooltipHtml={null}
          tooltipMode="mouse"
          tooltipOffset={
            Object {
              "left": 0,
              "top": -35,
            }
          }
          values={[Function]}
          width={350}
          x={[Function]}
          xScale={null}
          y={[Function]}
          y0={[Function]}
          yAxis={
            Object {
              "label": "Count",
              "tickArguments": Array [
                5,
              ],
            }
          }
          yScale={null}
        />
      </div>
      <div
        className=""
      >
        <h5>
          Airmass
        </h5>
        <GenericHistogram
          data={
            Array [
              undefined,
            ]
          }
          domain={
            Array [
              0,
              3,
            ]
          }
          height={150}
          id="airmass-hist"
          margin={60}
          nBins={36}
          nTicks={5}
          width={310}
          xLabel="Airmass [?]"
          yLabel="Count"
        />
      </div>
      <div
        className=""
      >
        <h5>
          Seeing
        </h5>
        <GenericHistogram
          data={
            Array [
              undefined,
            ]
          }
          domain={
            Array [
              0,
              2,
            ]
          }
          height={150}
          id="seeing-hist"
          margin={60}
          nBins={36}
          nTicks={5}
          width={310}
          xLabel="Seeing [?]"
          yLabel="Count"
        />
      </div>
      <div
        className=""
      >
        <h5>
          Sky brightness
        </h5>
        <GenericHistogram
          data={
            Array [
              undefined,
            ]
          }
          height={150}
          id="sky-brightness-hist"
          margin={60}
          nBins={36}
          nTicks={5}
          width={310}
          xLabel="Sky brightness [?]"
          yLabel="Count"
        />
      </div>
      <div
        className=""
      >
        <h5>
          Time baselines
        </h5>
        <GenericHistogram
          data={Array []}
          height={150}
          id="time-baselines-hist"
          logScale={true}
          margin={60}
          nBins={36}
          nTicks={5}
          width={310}
          xLabel="Time baseline [s]"
          yLabel="Count"
        />
      </div>
    </div>
  </div>
</Rnd>
`;

exports[`Fields details test renders correctly with shallowRender without fieldData 1`] = `
<Rnd
  className="field-details-popup"
  default={
    Object {
      "height": 580,
      "width": 1560,
      "x": 800,
      "y": 100,
    }
  }
  disableDragging={false}
  dragGrid={
    Array [
      20,
      20,
    ]
  }
  dragHandleClassName=".move-button"
  enableResizing={
    Object {
      "bottom": true,
      "bottomLeft": true,
      "bottomRight": true,
      "left": true,
      "right": true,
      "top": true,
      "topLeft": true,
      "topRight": true,
    }
  }
  maxHeight={9007199254740991}
  maxWidth={9007199254740991}
  onDrag={[Function]}
  onDragStart={[Function]}
  onDragStop={[Function]}
  onResize={[Function]}
  onResizeStart={[Function]}
  onResizeStop={[Function]}
  resizeGrid={
    Array [
      20,
      20,
    ]
  }
  resizeHandleClassName=".close-button"
  resizeHandleWrapperClass="field-details-popup"
>
  <div
    className="field-details"
  >
    <DraggableTitle
      customClass="field-details-title"
      onCloseClick={[Function]}
      showClose={true}
      title="Field ID: 0"
    />
    <div
      className="field-details-content"
    >
      <div
        className="summary-data"
      >
        <h5>
          Summary
        </h5>
        <div
          className="summary-data-content"
        >
          <div>
            FieldID: 
            0
          </div>
          <div>
            Timestamp: 
            Invalid Date Invalid Date
          </div>
          <div>
            Filter: 
            <FilterIndicator
              filterName={undefined}
            />
          </div>
          <div>
            Observations: 
            1
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <h5>
          Rotation angle
        </h5>
        <GenericHistogram
          data={
            Array [
              NaN,
            ]
          }
          domain={
            Array [
              0,
              360,
            ]
          }
          height={150}
          id="rotation-hist"
          margin={60}
          nBins={36}
          nTicks={5}
          width={310}
          xLabel="Rotation [deg]"
          yLabel="Count"
        />
      </div>
      <div
        className="histogram barchart"
      >
        <h5>
          Filters
        </h5>
        <Unknown
          barPadding={0.5}
          colorByLabel={false}
          colorScale={[Function]}
          data={
            Array [
              Object {
                "label": "filters",
                "values": Array [
                  Object {
                    "x": "u",
                    "y": 0,
                  },
                  Object {
                    "x": "g",
                    "y": 0,
                  },
                  Object {
                    "x": "r",
                    "y": 0,
                  },
                  Object {
                    "x": "i",
                    "y": 0,
                  },
                  Object {
                    "x": "z",
                    "y": 0,
                  },
                  Object {
                    "x": "y",
                    "y": 0,
                  },
                ],
              },
            ]
          }
          height={190}
          label={[Function]}
          margin={
            Object {
              "bottom": 30,
              "left": 30,
              "right": 20,
              "top": 20,
            }
          }
          offset="zero"
          order="default"
          tooltipContained={false}
          tooltipHtml={null}
          tooltipMode="mouse"
          tooltipOffset={
            Object {
              "left": 0,
              "top": -35,
            }
          }
          values={[Function]}
          width={350}
          x={[Function]}
          xScale={null}
          y={[Function]}
          y0={[Function]}
          yAxis={
            Object {
              "label": "Count",
              "tickArguments": Array [
                5,
              ],
            }
          }
          yScale={null}
        />
      </div>
      <div
        className="histogram barchart"
      >
        <h5>
          Alerts
        </h5>
        <Unknown
          barPadding={0.5}
          colorByLabel={false}
          colorScale={[Function]}
          data={
            Array [
              Object {
                "label": "somethingA",
                "values": Array [
                  Object {
                    "x": "Moving",
                    "y": 10,
                  },
                  Object {
                    "x": "Variable",
                    "y": 4,
                  },
                  Object {
                    "x": "Outburst",
                    "y": 1,
                  },
                  Object {
                    "x": "New",
                    "y": 5,
                  },
                  Object {
                    "x": "Vanished",
                    "y": 9,
                  },
                ],
              },
            ]
          }
          height={190}
          label={[Function]}
          margin={
            Object {
              "bottom": 30,
              "left": 30,
              "right": 20,
              "top": 20,
            }
          }
          offset="zero"
          order="default"
          tooltipContained={false}
          tooltipHtml={null}
          tooltipMode="mouse"
          tooltipOffset={
            Object {
              "left": 0,
              "top": -35,
            }
          }
          values={[Function]}
          width={350}
          x={[Function]}
          xScale={null}
          y={[Function]}
          y0={[Function]}
          yAxis={
            Object {
              "label": "Count",
              "tickArguments": Array [
                5,
              ],
            }
          }
          yScale={null}
        />
      </div>
      <div
        className=""
      >
        <h5>
          Airmass
        </h5>
        <GenericHistogram
          data={
            Array [
              undefined,
            ]
          }
          domain={
            Array [
              0,
              3,
            ]
          }
          height={150}
          id="airmass-hist"
          margin={60}
          nBins={36}
          nTicks={5}
          width={310}
          xLabel="Airmass [?]"
          yLabel="Count"
        />
      </div>
      <div
        className=""
      >
        <h5>
          Seeing
        </h5>
        <GenericHistogram
          data={
            Array [
              undefined,
            ]
          }
          domain={
            Array [
              0,
              2,
            ]
          }
          height={150}
          id="seeing-hist"
          margin={60}
          nBins={36}
          nTicks={5}
          width={310}
          xLabel="Seeing [?]"
          yLabel="Count"
        />
      </div>
      <div
        className=""
      >
        <h5>
          Sky brightness
        </h5>
        <GenericHistogram
          data={
            Array [
              undefined,
            ]
          }
          height={150}
          id="sky-brightness-hist"
          margin={60}
          nBins={36}
          nTicks={5}
          width={310}
          xLabel="Sky brightness [?]"
          yLabel="Count"
        />
      </div>
      <div
        className=""
      >
        <h5>
          Time baselines
        </h5>
        <GenericHistogram
          data={Array []}
          height={150}
          id="time-baselines-hist"
          logScale={true}
          margin={60}
          nBins={36}
          nTicks={5}
          width={310}
          xLabel="Time baseline [s]"
          yLabel="Count"
        />
      </div>
    </div>
  </div>
</Rnd>
`;
